var documenterSearchIndex = {"docs":
[{"location":"","page":"Home","title":"Home","text":"CurrentModule = PhyloPickMe","category":"page"},{"location":"#PhyloPickMe.jl-Documentation","page":"Home","title":"PhyloPickMe.jl Documentation","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"PhyloPickMe is a is a Julia package for selecting samples for inclusion in a Species Tree Analysis.","category":"page"},{"location":"#Context","page":"Home","title":"Context","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Biologists have access to vast amounts of DNA sequence data.  For a fixed collection of Species individual samples are usually missing from genetic data they have access to. So how does one determine when there is enough data to justify including a particular sample/species in one's analysis.  Typically, researchers have used ad hoc methodologies to make this decision (i.e. requiring sample be on 75% of all of the sequenced genes).  ","category":"page"},{"location":"","page":"Home","title":"Home","text":"Given a collection of estimated gene trees, PickMe is a software tool which recommends which of the samples should be included in a full analysis of the data based on Bayesian Analysis of whether or not the data can be described by a single species tree.  Thus PickMe can be a part of the species tree analysis pipeline.","category":"page"},{"location":"#Reference","page":"Home","title":"Reference","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"For both the PhyloPickMe package and the underlying sample selection methodology please cite:","category":"page"},{"location":"","page":"Home","title":"Home","text":"Joseph Rusinko, Yu Cai, Allison Doherty, Katherine Thompson, Julien Boutte, Mark Fishbein, and Shannon C. K. Straub (2024).   PickMe: Sample Selection for species tree reconstruction using coalescent weighted quartets   under review","category":"page"},{"location":"#Function-Usage","page":"Home","title":"Function Usage","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"PickMe(InputTreeFile::String, OutputFile::String)","category":"page"},{"location":"#PickMe-Tutorial-For-Julia-Beginners","page":"Home","title":"PickMe Tutorial For Julia Beginners","text":"","category":"section"},{"location":"#Installation","page":"Home","title":"Installation","text":"","category":"section"},{"location":"#Install-Julia","page":"Home","title":"Install Julia","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Use the Julia  Julia link to install Julia on your machine of choice.  PickMe itself is not dependent on your operating system.","category":"page"},{"location":"#Select-a-code-editor","page":"Home","title":"Select a code editor","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Select and install one of the following code editors.  (For R-natives, this is like installing R-Studio to run code written in the  R programming language.)","category":"page"},{"location":"","page":"Home","title":"Home","text":"Visual Studio Code provides an editor and an integrated development environment (IDE) for Julia: \nJuno provides an IDE for Julia, based on the Atom editor.\nA Jupyter notebook","category":"page"},{"location":"#Open-Julia-in-your-editor-of-choice.","page":"Home","title":"Open Julia in your editor of choice.","text":"","category":"section"},{"location":"#note-for-Julia","page":"Home","title":"note for Julia","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"When working with Julia you will see a command line that either says","category":"page"},{"location":"","page":"Home","title":"Home","text":"julia>","category":"page"},{"location":"","page":"Home","title":"Home","text":"This is the promt you want for running specific Julia Code.","category":"page"},{"location":"","page":"Home","title":"Home","text":"or","category":"page"},{"location":"","page":"Home","title":"Home","text":"    packageName>","category":"page"},{"location":"","page":"Home","title":"Home","text":"This is the prompt you want for loading or modifing Julia Packages.   You can toggle between the options using \"]\" to get to the package option a nd backspace to return to the Julia prompt.","category":"page"},{"location":"#add-the-Julia-Package","page":"Home","title":"add the Julia Package","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"First let tell Julia you want to use its package functions by running.","category":"page"},{"location":"","page":"Home","title":"Home","text":"julia> using Pkg","category":"page"},{"location":"","page":"Home","title":"Home","text":"Shift into package mode by toggling with \"]\".  Add the PickMe Package using ","category":"page"},{"location":"","page":"Home","title":"Home","text":"package> add PickMe","category":"page"},{"location":"","page":"Home","title":"Home","text":"You may need to click \"y\" to add the package, and wait for a couple of minutes as the program is setup.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Finally, toggle back to julia mode by pressing backspace.","category":"page"},{"location":"#Confirm-the-Julia-package-is-working","page":"Home","title":"Confirm the Julia package is working","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"To pre-compile the Julia code, and test that the program has successfully been installed run the code","category":"page"},{"location":"","page":"Home","title":"Home","text":"julia> PickMeJulia(\"sampletrees.txt\",\"Mytestoutput.csv\")","category":"page"},{"location":"","page":"Home","title":"Home","text":"The output in \"Mytesttouput.csv\" should match the \"testoutput.txt\" file on GitHub.","category":"page"},{"location":"#Running-Pick-Me","page":"Home","title":"Running Pick Me","text":"","category":"section"},{"location":"#Estimate-Initial-Gene-trees-using-all-available-data","page":"Home","title":"Estimate Initial Gene trees using all available data","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Estimate gene trees for your sequence data using your method of choice. (does not Require Julia). ","category":"page"},{"location":"#Input-files-requirements","page":"Home","title":"Input files requirements","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Trees should be collected into a single file, with one gene tree in Newick Format per line ending in with a;.","category":"page"},{"location":"","page":"Home","title":"Home","text":"PickMe and PickMeQ require a .txt file including all phylogenetic trees generated by phylogenomic tools (e.g. RAxML, IQ-TREE). All the trees files must have the same samples names (e.g. Sample_Potato). ","category":"page"},{"location":"","page":"Home","title":"Home","text":"Some gene tree pipelines, may include the gene identifier along with the sample in each of the individual gene trees.  PickMe will assume that each identifier/sample combination is its own unique sample. If this is the case, the .txt file can be cleaned using CTN.py.","category":"page"},{"location":"#Run-PickMe","page":"Home","title":"Run PickMe","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Assuming your treefile is in the active folder and is labeled \"genetrees.txt\" run the command","category":"page"},{"location":"","page":"Home","title":"Home","text":"PickMeJulia(\"genetrees.txt\",\"PicKmeOutputMyGeneTrees.csv\")","category":"page"},{"location":"#Begin-species-tree-analysis","page":"Home","title":"Begin species tree analysis","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"We recommend running your species tree analysis beginning with sequence alignment using only the samples identified by PickMe, as \"Good\",\"Very Good\", or \"Exceptional\".","category":"page"},{"location":"#Warnings","page":"Home","title":"Warnings","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"PickMe only uses the topology of each of the input gene trees.  It will safely ignore Bootstrap values, or branch lengths.  However we recommend  not estimating these values in your pre-PickMe analysis in order to save time. ","category":"page"},{"location":"","page":"Home","title":"Home","text":"PickMe assumes a single individual per species.  While PickMe will run with multiple samples per-species it will  treat each individual as its own species.  This is Not Recommended.","category":"page"},{"location":"","page":"Home","title":"Home","text":"PickMe should not be used as a test for hybridization, as hybrid-samples have the potential to be recommended by PickMe (see the Supplemental Material of the article for specific details).","category":"page"},{"location":"#Related-Packages","page":"Home","title":"Related Packages","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"The package PhyloNetworks may be useful tool for exploring non-tree based artificats of the set of gene trees, which PickMe is not designed to address.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Claudia Solís-Lemus, Paul Bastide and Cécile Ané (2017).   PhyloNetworks: a package for phylogenetic networks.   Molecular Biology and Evolution   34(12):3292–3298.   doi:10.1093/molbev/msx235","category":"page"},{"location":"#","page":"Home","title":"","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"This GitHub also contains an R-script which can be used to drop samples which are not recommended by PickMe, from a folder of DNA sequences.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Modules = [PhyloPickMe]","category":"page"},{"location":"#PhyloPickMe.PickMe-Tuple{String, String}","page":"Home","title":"PhyloPickMe.PickMe","text":"PickMe(InputTreeFile::String, OutputFile::String)\n\nClassify all samples which occur in the InputTreeFile using PickMe.\n\n\n\n\n\n","category":"method"}]
}
